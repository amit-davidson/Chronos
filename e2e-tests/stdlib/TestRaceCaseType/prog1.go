package main

func main() {
	var x, y int
	var i interface{} = x
	c := make(chan int, 1)
	go func() {
		switch i.(type) {
		case nil:
		case int:
		}
		c <- 1
	}()
	i = y
	<-c
}
