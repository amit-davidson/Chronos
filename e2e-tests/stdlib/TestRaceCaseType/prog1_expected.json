{
	"Lockset": {
		"ExistingLocks": {},
		"ExistingUnlocks": {}
	},
	"GuardedAccess": [
		{
			"ID": 1,
			"PosID": 1,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:5:6",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 1
				}
			}
		},
		{
			"ID": 2,
			"PosID": 2,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:5:6",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:5:6",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				}
			}
		},
		{
			"ID": 3,
			"PosID": 3,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:6:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:6:11",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 3
				}
			}
		},
		{
			"ID": 4,
			"PosID": 4,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:6:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:6:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 4
				}
			}
		},
		{
			"ID": 5,
			"PosID": 5,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:8:10",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:5:6",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 1
				}
			}
		},
		{
			"ID": 6,
			"PosID": 6,
			"Pos": "",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:8:10",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 2
				}
			}
		},
		{
			"ID": 7,
			"PosID": 7,
			"Pos": "",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 3
				}
			}
		},
		{
			"ID": 8,
			"PosID": 8,
			"Pos": "",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 4
				}
			}
		},
		{
			"ID": 9,
			"PosID": 9,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:12:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:6:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 5
				}
			}
		},
		{
			"ID": 10,
			"PosID": 10,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:10:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:8:10",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 6
				}
			}
		},
		{
			"ID": 11,
			"PosID": 11,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:14:2",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 6
				}
			}
		},
		{
			"ID": 12,
			"PosID": 12,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:14:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:5:6",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 7
				}
			}
		},
		{
			"ID": 13,
			"PosID": 13,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:15:4",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:6:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 8
				}
			}
		},
		{
			"ID": 14,
			"PosID": 14,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:15:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestRaceCaseType/prog1.go:15:4",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 9
				}
			}
		}
	]
}